### PR TITLE
feat(cloudwatch): explicit, readable alarm names with override seam

### PR DESCRIPTION
**Pull request by laazyj**
## Summary

Auto-generated alarms now ship with an explicit `AlarmName` derived from the stack name, builder id, and alarm key (kebab-cased, slash-joined). Operators see hierarchical names in the CloudWatch console instead of CFN-generated hash suffixes.

**Before:** `jasonduffettnetsiteAlertsNumberOfNotificationsFailedAlarmF37A3C89`
**After:** `jason-duffett-net-site-stack/site-alerts/number-of-notifications-failed`

## Override seams (layered, lowest precedence first)

1. **Library default** — `defaultAlarmName(scope, id, key)` from `${stackName}/${kebab(id)}/${kebab(key)}`.
2. **Per-alarm** — `recommendedAlarms: { errors: { alarmName: alarmName(\"checkout-fn-errors\") } }` and `addAlarm(...).alarmName(...)`. The `AlarmName` brand forces a one-line opt-in via the `alarmName()` helper, which validates charset (CloudWatch allows `[A-Za-z0-9-_./#:()+ =@]`) and length (<= 255).
3. **Cross-cutting policy** — `alarmNamePolicy(scope, { defaults, rules })` mirrors the existing `alarmActionsPolicy`. Supports `prefix` / `suffix` / `transform` decorations with substring, regex, and predicate matchers.

```ts
alarmNamePolicy(app, {
  defaults: { prefix: \"prod\" },
  rules: [
    { match: /Errors\$/, suffix: \"critical\" },
    { match: \"throttles\", suffix: \"warning\" },
    { match: ctx => ctx.path.includes(\"payments\"), prefix: \"payments\" },
  ],
});
```

## Threading

- `AlarmDefinition.alarmName?: AlarmName` and `AlarmConfig.alarmName?: AlarmName` carry the override through the existing `resolve* / createAlarms` pipeline (one-line propagation in each `*-alarms.ts`).
- New `AlarmConfigDefaults = Required<Omit<AlarmConfig, \"alarmName\">>` keeps per-package `*_ALARM_DEFAULTS` typing tight without forcing a name into defaults.
- `createAlarms` resolves `def.alarmName ?? defaultAlarmName(scope, id, def.key)`. Construct ids are unchanged, so `alarmActionsPolicy` matchers and existing logical ids stay stable.

## Test plan

- [x] New unit tests in `packages/cloudwatch/test/`:
  - `alarm-name.test.ts` — `alarmName()` validation, `kebab()` edge cases, `joinAlarmName()` separator + empty-segment handling.
  - `default-alarm-name.test.ts` — `${stackName}/${kebab(id)}/${kebab(key)}` formatting.
  - `alarm-name-policy.test.ts` — defaults, layering, transforms, separator, regex matching, `replaceDefaults`, no-name passthrough, invalid-char rejection.
  - `create-alarms.test.ts` — default-name fallback and explicit override.
  - `alarm-definition-builder.test.ts` — `.alarmName()` setter.
- [x] `packages/lambda/test/function-alarms.test.ts` — verifies recommendedAlarms override and addAlarm threading.
- [x] All 6 examples-package snapshots refreshed (`+38` lines, all `AlarmName` additions; no other CFN drift).
- [x] `npm run lint && npm run format:check` clean.
- [x] `npx nx run-many -t test` — 13 projects, all pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by laazyj** [comment]
Not sure about the kebab-casing in the defaultAlarmName. It greatly increases the length of the alarm name without adding much readability.

File: packages/cloudwatch/src/index.ts [hunk 1]
```diff
@@ -1,10 +1,19 @@
-export type { AlarmConfig } from "./alarm-config.js";
+export type { AlarmConfig, AlarmConfigDefaults } from "./alarm-config.js";
 export type { AlarmDefinition } from "./alarm-definition.js";
 export { AlarmDefinitionBuilder } from "./alarm-definition-builder.js";
+export { type AlarmName, alarmName, joinAlarmName, kebab } from "./alarm-name.js";
```
Comment (laazyj) [inline packages/cloudwatch/src/index.ts]: Why export kebab and joinAlarmName? Are these intended for use outside of the package?

File: packages/cloudwatch/src/policies/policy-matcher.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  if (matcher instanceof RegExp) return matcher.test(ctx.path);
+  return ctx.id.includes(matcher) || ctx.path.includes(matcher);
+}
+
+export function ruleMatches(rule: AlarmRuleScope, ctx: AlarmMatchContext): boolean {
+  if (rule.singleOnly === true && ctx.isComposite) return false;
+  if (rule.compositeOnly === true && !ctx.isComposite) return false;
+  const matchers = Array.isArray(rule.match) ? rule.match : [rule.match];
+  return matchers.some((m) => matchesOne(m, ctx));
+}
```
Comment (laazyj) [inline packages/cloudwatch/src/policies/policy-matcher.ts:44]: Where are the unit tests for policy-matcher.ts?

**Comment by laazyj** [comment]
Where are the cloudwatch/README updates? These are important to advertise the new policy

**Comment by laazyj** [comment]
Push back on kebab case naming is pretty compelling:

- NumberOfNotificationsFailed → number-of-notifications-failed. That's 28 → 30 chars. kebab adds one character per word boundary, typical alarm names have 3–6 boundaries.
- Readability cuts the other way for scan tasks. PascalCase compresses well in code where you're already in editor flow, but alarm names are read in dashboards, email subjects, oncall pages, and aws cloudwatch describe-alarms output. In those contexts hyphens are objectively faster to parse than uppercase boundaries — same reason URL slugs and k8s
  resource names converged on kebab.
- The before/after in the PR body (`jasonduffettnetsiteAlertsNumberOfNotificationsFailedAlarmF37A3C89` →
  `jason-duffett-net-site-stack/site-alerts/number-of-notifications-failed`) is the strongest argument here, and it's the user-visible win of the whole PR. Drop the kebab and you lose most of it.
- 255-char limit isn't materially threatened by the hyphens.

If you want shorter names, the lever is the segments (shorter stack name, shorter id), not the casing.
